app/models/ticket.rb

```diff
@@ -22,6 +22,10 @@ def action_allowed?(action, user)
     end
   end
 
+  def can_user_access?(user)
+    user_stakeholders(user).any?
```

Comment:
I vaguely remember that you did a similar check in other, already-existing controller methods.
Can you go back and check whether you can re-use this new method in other places, now that you've created it?

Reply:
Also, I want to note that this query can become very expensive very quickly. In a lot of developers' minds (who aren't familiar with the in-depth implementation for your tickets) the intuition is that "a simple boolean method that gives a Yes-No answer surely will just to some simple `==` comparison(s)". But the truth is that even in this early stage, the `user_stakeholders` already does a lot of heavy lifting, loading roles and stuff under the hood.

So _at least_ a warning comment should be warranted, and potentially we even need to think about a better optimization.

Reply:
The PR is merged, so removed this method. (looks like I forgot to reply the previous message, it was still in draft. Sending all of them together now)